### PR TITLE
Generic pom.xml cleanups

### PR DESCRIPTION
**Pull request by jkosternl**
Nothing to do with Java 11, so should be fine to merge 😉

### CODE REVIEW

File: credentialProvider/pom.xml [hunk 1]
```diff
@@ -96,13 +96,6 @@
...
 		</dependency>
 
-		<!-- commons-collections 3.2.2 included to override 3.2.1 version included in apacheds-test-framework -->
-		<dependency>
-			<groupId>commons-collections</groupId>
-			<artifactId>commons-collections</artifactId>
-			<version>3.2.2</version>
-			<scope>test</scope>
-		</dependency>
 		<dependency>
```
Comment (jkosternl) [inline credentialProvider/pom.xml:99]: This is not needed anymore. Already on 3.2.2 without these lines atm.

File: iaf-commons/pom.xml [hunk 2]
```diff
@@ -13,12 +13,6 @@
 	<artifactId>ibis-adapterframework-commons</artifactId>
 	<name>Ibis AdapterFramework Commons</name>
 
-	<properties>
-		<maven.compiler.source>8</maven.compiler.source>
-		<maven.compiler.target>8</maven.compiler.target>
-		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-	</properties>
-
```
Comment (jkosternl) [inline iaf-commons/pom.xml:21]: Already inherited from the parent pom, so this is duplicate.

File: idin/pom.xml [hunk 3]
```diff
@@ -35,9 +35,7 @@
 		</dependency>
 		<dependency>
 			<groupId>org.mockito</groupId>
-			<artifactId>mockito-all</artifactId>
-			<version>1.10.19</version>
-			<scope>test</scope>
+			<artifactId>mockito-core</artifactId>
```
Comment (jkosternl) [inline idin/pom.xml:38]: mockito-all is very old. Current version is `mockito-core`

File: aspose/pom.xml [hunk 4]
```diff
@@ -25,13 +25,6 @@
 			<artifactId>lombok</artifactId>
 		</dependency>
 
-		<!-- Explicit dependency to mitigate CVE-2022-3171 -->
-		<dependency>
-			<groupId>com.google.protobuf</groupId>
-			<artifactId>protobuf-java</artifactId>
-			<version>3.24.3</version>
-		</dependency>
```
Comment (jkosternl) [inline aspose/pom.xml:33]: Should be moved to parent pom, since somewhere else there is still a transitive dep found, using the older version...

**Review by nielsm5 — APPROVED** [review]

**Review by tnleeuw — APPROVED** [review]

**Pull request merged**